core/src/main/java/org/elasticsearch/search/aggregations/AggregatorFactory.java

```diff
@@ -54,6 +62,19 @@ public AggregatorFactory(String name, String type) {
     }
 
     /**
+     * Initializes this factory with the given {@link AggregationContext} ready
+     * to create {@link Aggregator}s
+     */
+    public final void init(AggregationContext context) {
+        this.context = context;
+        doInit(context);
+        this.factories.init(context);
+    }
+
+    protected void doInit(AggregationContext context) {
```

Comment:
can you add javadocs saying what kind of initialization steps need to be performed here?